### PR TITLE
Adding function match for regex based conditions.

### PR DESCRIPTION
**Pull request by karheinz**
The first commit adds a function called 'match' for regex based conditions.
The second commit fixes a typo in the docs.

### CODE REVIEW

File: test/test_input_recipe.py [hunk 1]
```diff
@@ -149,3 +149,29 @@ def testNestedFwd(self):
         self.cmpEntry(res[3], "d", fwd=True)
         self.cmpEntry(res[4], "e")
 
+    def testConditions(self):
```
Comment (jkloetzke) [inline test/test_input_recipe.py]: That unit test looks odd. If at all it should be in a separate test suite and in a file called `test_input_env.py`. All these cases simply test the `funMatch` function that is already tested in `test_input_stringparser.py`. Instead a dedicated test for bob.input.Env.evaluate should be created without relying on a particular string function.

**Comment by karheinz** [comment]
OK, I removed the test.

**Comment by jkloetzke** [comment]
Looks good. Could you please squash the last commits with the previous?

**Comment by karheinz** [comment]
Done ;)

**Comment by rhubert** [comment]
I think someone should increment bob version after this commit...

**Comment by jkloetzke** [comment]
Thanks. I would increment the version number with the next release. Otherwise we have to increment it basically with every commit...

**Comment by rhubert** [comment]
I don't think so. Someone, lets call him Philipp, may want to use `match` tomorrow and in this case he needs to make sure the other team members update to a bob version supporting `match` using `bobMinimumVersion`.  :) 

Other commits (like `bob status`) do not need a new bob version...

**Comment by jkloetzke** [comment]
Good point. I've pushed v0.7.0 with the changes so far...

**Pull request merged**